appcues/src/main/java/com/appcues/statemachine/StateMachine.kt

```diff
@@ -102,11 +104,25 @@ internal class StateMachine(
                     Failure(sideEffect.error)
                 }
                 is PresentContainerEffect -> {
-                    // kick off UI
-                    sideEffect.experience.stepContainers[sideEffect.containerIndex].presentingTrait.present()
-
-                    // wait on the RenderingStep state to flow in from the UI, return that result
-                    sideEffect.completion.await()
+                    try {
+                        // kick off UI
+                        sideEffect.experience.stepContainers[sideEffect.containerIndex].presentingTrait.present()
+                        // wait on the RenderingStep state to flow in from the UI, return that result
+                        sideEffect.completion.await()
+                    } catch (exception: AppcuesTraitException) {
+                        // force state machine to move to idling when we get this exception
+                        _state = Idling
```

Comment:
this part we need to think about a bit. I think it needs to not just set to Idling, but actually do a proper transition through the ending of the experience, for analytics purposes. We may need to call back into `handleAction` (like the ContinuationEffect does above) - with an `EndExperience` action, or maybe `Reset`? I need to understand the expected analytics behavior when say, step 3 of 6 fails to present, and the experience is killed.

Reply:
on iOS, this type of fatal error gets processed here https://github.com/appcues/appcues-ios-sdk/blob/main/Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine.swift#L264, which does end up setting the machine right back to idling https://github.com/appcues/appcues-ios-sdk/blob/main/Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine.swift#L237 -- so maybe this _is_ just fine.

@mmaatttt double checking, do you recall if there was any prior discussion here on errors? Is it expected that no experience_dismissed or any other step events would be triggered, if an experience has a fatal error? thanks

Reply:
Yeah, I don't recall how extensively we talked about it, but to me, the idea of a fatal error is you get that and nothing else.

Reply:
can we leave as is and iterate back if we need to ?

Reply:
yes, sorry just wanted to think that one through, but as long as we are consistent, and I didn't see anything in the [event model docs](https://www.notion.so/appcues/Experience-event-model-a05021ffec6c40f091ac01b5e9f94bc4) stating otherwise - then this seems good